cloud/services/subnets/subnets.go

```diff
@@ -35,12 +35,29 @@ func (s *Service) getExisting(ctx context.Context, rgName string, spec azure.Sub
 		return nil, errors.Wrapf(err, "failed to fetch subnet named %s in vnet %s", spec.VNetName, spec.Name)
 	}
 
+	IPV4Cidr := to.String(subnet.SubnetPropertiesFormat.AddressPrefix)
+	var IPV6Cidr string
+	if subnet.SubnetPropertiesFormat != nil && subnet.SubnetPropertiesFormat.AddressPrefixes != nil {
+		addresses := to.StringSlice(subnet.SubnetPropertiesFormat.AddressPrefixes)
+
+		// assumption that the second address is the ipv6 address.
+		// This would be unnessary if subnet spec was array
+		if len(addresses) > 0 {
```

Comment:
Is this safe to assume? Or should we determine the IP family by parsing the CIDR -  https://github.com/kubernetes/utils/blob/master/net/net.go#L127-L132

Reply:
Ideally we don't have these checks at all.  If we are able to move the api model to list of cidr blocks [as documented for a change to v1alpha4](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/618#issuecomment-657821311), this could be removed completely.

Reply:
removed